lib/streamlit/util.py

```diff
@@ -126,6 +128,11 @@ def index_(iterable, x) -> int:
     """
 
     for i, value in enumerate(iterable):
+        # https://stackoverflow.com/questions/588004/is-floating-point-math-broken
+        # https://github.com/streamlit/streamlit/issues/4663
+        if isinstance(iterable, np.ndarray):
```

Comment:
Whoops, so I just realized that I don't think this is a full fix for the issue.

Floating point precision errors are going to happen with `float` types in general, not just those in numpy arrays.

Here's a test case that can be added to the parametrized tests that will break when it's supposed to work: `(list(np.arange(0.0, 0.25, 0.05)), 0.15, 3)`